bigquery_etl/shredder/delete.py

```diff
@@ -285,13 +296,24 @@ class Partition:
     condition: str
     id: Optional[str] = None
     is_special: bool = False
+    required_partition_filter: Optional[str] = None
 
 
 def get_partition(table, partition_expr, end_date, id_=None) -> Optional[Partition]:
     """Return a Partition for id_ unless it is a date on or after end_date."""
     if id_ is None:
         if table.time_partitioning:
-            return Partition(condition=f"{partition_expr} < '{end_date}'")
+            return Partition(
+                condition=f"{partition_expr} < '{end_date}'",
+                required_partition_filter=(
+                    # explicitly select all partitions
+                    f"{partition_expr} IS NULL"
+                    f" OR {partition_expr} < CURRENT_DATE"
+                    f" OR {partition_expr} >= CURRENT_DATE"
+                )
+                if table.require_partition_filter
+                else None,
```

Comment:
IMO this would be more easily understandable if the `if`/`else` were within the parentheses.
```suggestion
                required_partition_filter=(
                    # explicitly select all partitions
                    f"{partition_expr} IS NULL"
                    f" OR {partition_expr} < CURRENT_DATE"
                    f" OR {partition_expr} >= CURRENT_DATE"
                    if table.require_partition_filter
                    else None,
                )
```